// fjsmthlib/lib/fjsmth.js
/**
 * Created by chenlizan on 15-3-18.
 */

var util = require("util");
var eventEmitter = require("events").EventEmitter;
var NativeFJ = require(__dirname + '/../build/Release/Fjsmth.node').Fjsmth_clz;

function GenCallbkFn(devid, msgcode, data, datalen) {
    myThis.emit('GenCallbk', devid, msgcode, data, datalen);
}

function FJsmth() {
    var self = this;
    self._native = new NativeFJ();
    self._native.SetAppTcpCallbkFn(GenCallbkFn);
}
util.inherits(FJsmth, eventEmitter);

FJsmth.prototype.StartLibWork = function (connection) {
    this._native.StartLibWork(connection);
}

FJsmth.prototype.GetServerNetStatus = function (cb) {
    cb(this._native.GetServerNetStatus());
}

FJsmth.prototype.AddUserDevice = function (DevId, cb) {
    cb(this._native.AddUserDevice(DevId));
}

FJsmth.prototype.ReadDevStateData = function (DevId, ParamBuff, cb) {
    cb(this._native.ReadDevStateData(DevId, ParamBuff));
}

FJsmth.prototype.ReadDevSingleStateData = function (args, cb) {
    cb(this._native.ReadDevSingleStateData(args));
}

module.exports.FJsmth = function () {
    return myThis = new FJsmth();
}
